extensions/functions_string.yaml

```diff
@@ -473,6 +498,138 @@ scalar_functions:
           - value: "string"
             name: "substring"
             description: The substring to search for.
+        return: i32
+      - args:
+          - name: case_sensitivity
+            options: [ CASE_SENSITIVE, CASE_INSENSITIVE, CASE_INSENSITIVE_ASCII ]
+            required: false
+          - value: "string"
+            name: "input"
+            description: The input string.
+          - value: "varchar<L1>"
+            name: "substring"
+            description: The substring to search for.
+        return: i32
+      - args:
+          - name: case_sensitivity
+            options: [ CASE_SENSITIVE, CASE_INSENSITIVE, CASE_INSENSITIVE_ASCII ]
+            required: false
+          - value: "string"
+            name: "input"
+            description: The input string.
+          - value: "fixedchar<L2>"
+            name: "substring"
+            description: The substring to search for.
```

Comment:
The original MR kept the types consistent. Is there a particular reason you want to introduce the interaction between different types?

Here's the original comment about this: https://github.com/substrait-io/substrait/issues/251#issuecomment-1191249269

Reply:
Thanks for your comments, and I kept the types consistent.